clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10350,33 +10350,50 @@ void OffloadWrapper::ConstructJob(Compilation &C, const JobAction &JA,
   assert(JA.getInputs().size() == Inputs.size() &&
          "Not have inputs for all dependence actions??");
 
-  // For FPGA, we wrap the host objects before archiving them when using
-  // -fsycl-link.  This allows for better extraction control from the
-  // archive when we need the host objects for subsequent compilations.
   if (OffloadingKind == Action::OFK_None &&
-      C.getArgs().hasArg(options::OPT_fintelfpga) &&
       C.getArgs().hasArg(options::OPT_fsycl_link_EQ)) {
 
-    // Add offload targets and inputs.
-    CmdArgs.push_back(C.getArgs().MakeArgString(
-        Twine("-kind=") + Action::GetOffloadKindName(OffloadingKind)));
-    CmdArgs.push_back(
-        TCArgs.MakeArgString(Twine("-target=") + Triple.getTriple()));
+    // When compiling and linking separately, we need to propagate the
+    // compression related CLI options to offload-wrapper. Don't propagate
+    // these options when wrapping objects for FPGA.
+    if (!C.getArgs().hasArg(options::OPT_fintelfpga)) {
```

Comment:
Code style nit.
When if-statement have both statement-true and statement-false, the code more readable when if-condition doesn't use negation.
I.e. prefer
```c++
if (cond) 
  A;
else
  B;
```
to
```c++
if (!cond) 
  B;
else
  A;
```

Reply:
Fixed in 20b31b75a96a37ee56ae279f79bfdfd4bb1a0388